src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -40,7 +57,9 @@ class PySnapshot:
     def files(self) -> Tuple[str, ...]: ...
 
 class PyExecutionRequest:
-    def __init__(self, **kwargs: Any) -> None: ...
+    def __init__(
+        self, poll: bool, poll_delay_in_ms: float | None, timeout_in_ms: float | None
```

Comment:
On the rust side you have these declared as `Option<u64>`, which is `int | None`.